http_server.c

```diff
@@ -166,18 +167,26 @@ static int http_server_worker(void *arg)
     }
 
     request.socket = socket;
+    // 設定 parser 初始參數
     http_parser_init(&parser, HTTP_REQUEST);
     parser.data = &request;
+
+    // 判斷執行緒是否該被中止
     while (!kthread_should_stop()) {
+        // 接收資料
         int ret = http_server_recv(socket, buf, RECV_BUFFER_SIZE - 1);
         if (ret <= 0) {
             if (ret)
                 pr_err("recv error: %d\n", ret);
             break;
         }
+
+        // 解析收到的資料
         http_parser_execute(&parser, &setting, buf, ret);
         if (request.complete && !http_should_keep_alive(&parser))
             break;
+
+        memset(buf, 0, ret);
```

Comment:
better to move this to the beginning of while loop, because kmalloc doesn't set memory to zero.

Reply:
OK, thanks!